### PR TITLE
Moved examples out of test

### PR DESCRIPTION
**Pull request by roxx30198**
- Moved the examples folder out from tests folder of numba_dpex.
- There are tests inside tests folder that are currently dependent on examples. Is it required to remove those tests, or move the files required inside the test folder itself?

### CODE REVIEW

File: numba_dpex/tests/debugging/common.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 def script_path(script):
     package_path = pathlib.Path(numba_dpex.__file__).parent
-    return str(package_path / "examples/debug" / script)
+    return str(package_path / "../examples/debug" / script)
```
Comment (diptorupd) [inline numba_dpex/tests/debugging/common.py:16]: The change will not work as is. Once you move the examples out of tree, any one who installs the package will not be able to run tests from the package as the example path will be invalid.

I suggest moving whatever source files that the debug tests use from the `examples/debug` into the `tests/debugging` directory.

File: numba_dpex/tests/test_examples.py [hunk 2]
```diff
@@ -11,6 +11,6 @@
 
 def test_examples_available():
     package_path = os.path.dirname(numba_dpex.__file__)
-    examples_path = os.path.join(package_path, "examples")
+    examples_path = os.path.join(package_path, "../examples")
```
Comment (diptorupd) [inline numba_dpex/tests/test_examples.py:14]: Same issue as the debugging test. Once the tests are out of tree, I suggest we remove this particular unit test completely.

File: numba_dpex/tests/debugging/common.py (hunk 1 above)
Comment (roxx30198) [reply numba_dpex/tests/debugging/common.py]: Moved files to tests/debugging and updated path.

File: numba_dpex/tests/test_examples.py (hunk 2 above)
Comment (roxx30198) [reply numba_dpex/tests/test_examples.py]: Removed tests_examples.py file from numba_dpex/tests.

**Pull request closed without merging**